dev-lang/fbc/fbc-1.06.0.ebuild

```diff
@@ -0,0 +1,89 @@
+# Copyright 1999-2019 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit eutils
+
+DESCRIPTION="FreeBASIC - A free/open source, multi-platform BASIC compiler."
+HOMEPAGE="http://www.freebasic.net"
+SRC_URI="https://github.com/freebasic/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz
+	https://github.com/freebasic/${PN}/releases/download/${PV}/FreeBASIC-${PV}-source-bootstrap.tar.xz"
+
+LICENSE="FDL-1.2 GPL-2+ LGPL-2.1+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~x86"
+IUSE="gpm libffi opengl X"
+
+DEPEND="
+	sys-libs/ncurses
+	gpm? ( sys-libs/gpm )
+	libffi? ( virtual/libffi )
+	opengl? ( virtual/opengl )
+	X? (
+		x11-libs/libX11
+		x11-libs/libXext
+		x11-libs/libXpm
+		x11-libs/libXrandr
+		x11-libs/libXrender
+	)"
+RDEPEND="${DEPEND}"
+
+DOCS="${S}/doc/fbc.1"
+
+BOOTSTRAP_S="${WORKDIR}/FreeBASIC-${PV}-source-bootstrap"
+
+src_unpack() {
+	# We only need bootstrap source code if fbc is not already present
+	if ! has_version dev-lang/fbc; then
+		unpack FreeBASIC-${PV}-source-bootstrap.tar.xz
+	fi
+	unpack ${P}.tar.gz
+}
+
+src_prepare() {
+	# We only need bootstrap source code if fbc is not already present
+	if ! has_version dev-lang/fbc; then
+		cd ${BOOTSTRAP_S}
```

Comment:
All external commands need to be appended with `|| die` because if executing them fails somehow, the emerging will not continue as wished. You have many `cd` commands present without `|| die` in here.